### DISCUSSION TITLE
Setting Form field value to Base64 Image string throwing an error.

### DISCUSSION THREAD

**Discussion by dakshsaraf** [react-hook-form/react-hook-form#2496] [Q&A] [answered]
I am trying to set the form field value to a base64 image string and it is throwing an error.
![image](https://user-images.githubusercontent.com/25619616/89509633-2cf9a280-d7ed-11ea-804a-36c801cfa5e0.png)

Here is a link to the code sandbox
https://codesandbox.io/s/react-hook-form-renders-umzuv?file=/src/ImageUpload.js

Any help on this matter will be greatly appreciated. I believe this could be an issue with RHF as the same implementation was working with formik earlier

**Comment by bluebill1049** [comment, marked as answer]
1. wrong usage with Controller: https://react-hook-form.com/api#Controller 
2. input file is immutable if you want to transform file data, it's better to sore in a custom register input.

```
useEffect(() => {
  register('inputData')
}, [inputData]);


setValue('inputData', yourImage)
```

**Reply by dakshsaraf** [reply]
Can you elaborate a bit more as to why this is a wrong usage of Controller?

**Reply by bluebill1049** [reply]
<img width="586" alt="Screen Shot 2020-08-06 at 7 08 47 pm" src="https://user-images.githubusercontent.com/10513364/89513723-43692380-d818-11ea-8cab-ca203b2a5e26.png">

have a reed on the doc. render provides methods which you shouldn't setValue manually.

**Reply by dakshsaraf** [reply]
1. If I use a custom register and do setValue manually. Do I need to perform setBlur manually too?
2. Can I create a custom register to pass to the inputRef argument because I wanted to perform .click() from a differnet DOM element
2. I have 2 other components where I use Controller and still setValue. Should I refactor them to use custom registers (they currently work)

**Reply by bluebill1049** [reply]
1, yes, `onBlur={setValue('xxx', 'whatever')}` if you should to validate you can turn on `shouldValidate: true`
2, what do you mean by a different DOM?
3. when using `Controller` you should never use `setValue`, let the `Controller` do its work.

**Reply by bluebill1049** [reply]
have a look this CSB: https://codesandbox.io/s/react-hook-form-v6-controller-qsd8r

**Reply by dakshsaraf** [reply]
By second question I mean that I want to trigger the click of the input from a different button so I would need to pass a ref to the inputRef, can I just pass the custom register and then do customRegister.current.click()

**Reply by bluebill1049** [reply]
oh: https://react-hook-form.com/api#trigger

**Reply by dakshsaraf** [reply]
I did not mean that, I guess that is a question for a different discussion post. I am fundamentally confused about one thing, I will try to frame my question better this time.
![image](https://user-images.githubusercontent.com/25619616/89516416-0a1fbc00-d7f6-11ea-9334-5835fe270eec.png)
In the above image a custom register is used and it is not passed to the **ref property of InputField** (can I do that?). If not, I just want to confirm my understanding is right. I need to set both onchange and onBlur like this {setValue('xxx', 'whatever')}

**Reply by bluebill1049** [reply]
> In the above image a custom register is used and it is not passed to the ref property of InputField (can I do that?) 

yes, you can.

> I need to set both onchange and onBlur like this {setValue('xxx', 'whatever')}

It depend on your use case, if you need both event to update the state then yes. you should attach `onBlur` and `onChange` with `setValue`

**Reply by dakshsaraf** [reply]
Thanks

**Reply by kartikchitade** [reply]
![register](https://user-images.githubusercontent.com/68079408/135730350-275aff91-d430-4662-9f4b-9c55bf878894.png)
how to validate or register this field in react hook form
